docs/advanced/data_spec.md

```diff
@@ -9,8 +9,64 @@ DeepGNN supports both homogeneous and heterogeneous graphs. Nodes and Edges supp
 
 DeepGNN supports two file formats: JSON and TSV. Users can generate a graph in either format then our pipeline will convert it into binary for training and inference.
 
-1. [JSON](#json-format): Heterogeneous or homegeneous graph.
-2. [TSV](#tsv-format): Homogeneous graph only.
+1. [Linear](#linear-format): Heterogeneous or homegeneous graph.
+2. [JSON](#json-format): Heterogeneous or homegeneous graph.
+3. [TSV](#tsv-format): Homogeneous graph only.
+
+## Linear Format
+
+Here is the graph data Linear format. The format requires two files: graph.linear and meta.json.
+
+### Graph Data
+
+`graph.linear` layout
+
+```
+<node info> <edge_1_info> <edge_2_info> ...
+```
+  node_info: -1 node_id node_type node_weight <features>
```

Comment:
why src = -1 is the flag for node lines, rather than dst = -1?

Reply:
I think the code is cleaner like this and easier to parse